src/encryptsave/mod.rs

```diff
@@ -0,0 +1,175 @@
+mod ll;
+pub mod errors;
+
+
+macro_rules! tox_res {
+    ( $res:ident <- $rexpr:expr, $err:ident, $r:expr ) => {
```

Comment:
This code, `tox_res!` do not know who is `$res`(`pk`), can't return to `Ok($res)`(`Ok(pk)`).

Macros are used to avoid duplication of effort, I can remove all macros.